core/station/impl/src/services/system.rs

```diff
@@ -136,13 +132,19 @@ impl SystemService {
     }
 
     /// Execute an upgrade of the station by requesting the upgrader to perform it on our behalf.
-    pub async fn upgrade_station(&self, module: &[u8], arg: &[u8]) -> ServiceResult<()> {
+    pub async fn upgrade_station(
+        &self,
+        request_id: UUID,
+        module: &[u8],
+        arg: &[u8],
+    ) -> ServiceResult<()> {
         let upgrader_canister_id = self.get_upgrader_canister_id();
 
         ic_cdk::call(
             upgrader_canister_id,
             "trigger_upgrade",
-            (InstallCanisterParams {
+            (UpgradeParams {
+                request_id: Uuid::from_bytes(request_id).hyphenated().to_string(),
```

Comment:
the upgrade request id is already stored in the station, under `https://github.com/dfinity/orbit/blob/main/core/station/impl/src/models/system.rs#L47` instead of adding this in the UpgradeParams the above `notify_failed_station_upgrade` method would just need to receive the `reason`

Reply:
Good point! Fixed.